auth/src/main/java/com/firebase/ui/auth/ui/idp/CredentialSignInHandler.java

```diff
@@ -82,9 +84,10 @@ public void onComplete(@NonNull Task<AuthResult> task) {
                             .addOnFailureListener(new OnFailureListener() {
                                 @Override
                                 public void onFailure(@NonNull Exception e) {
-                                    // TODO: What to do when signing in with Credential fails
-                                    // and we can't continue to Welcome back flow without
-                                    // knowing providers?
+                                    mHelper.finishActivity(
+                                            mActivity,
+                                            ResultCodes.CANCELED,
+                                            IdpResponse.getErrorCodeIntent(ErrorCodes.UNKNOWN_ERROR));
```

Comment:
Agree, having some solution is better here.  When you say "refactor the ErrorCodes" what do you have in mind?  I'd say 'UNKNOWN_ERROR' is actually somewhat fitting here since this is a corner case situation.